components/mdns/mdns.c

```diff
@@ -5956,11 +5956,6 @@ static mdns_txt_item_t *_copy_mdns_txt_items(mdns_txt_linked_item_t *items, uint
     for (mdns_txt_linked_item_t *tmp = items; tmp != NULL; tmp = tmp->next) {
         ret_index++;
     }
-    if (ret_index == 0) {
```

Comment:
Isn't this covering the case for `items==NULL`? Since it is a static function, I expect that this case is covered in the call site.

Reply:
It's not covered by the calling party (I think correctly) due to hierarchical reasons, as we can call this multiple times to make copies of txt(s) without needing to check for `NULL`.

Anyway, this fix is actually removing the redundant fix, which is already there, just few lines below:

https://github.com/espressif/esp-protocols/blob/25b3d5fd7b1d42ba52c8faf819080ebec9af860b/components/mdns/mdns.c#L5959-L5963

(fixing merge conflict)